scripts/download_modis.py

```diff
@@ -0,0 +1,165 @@
+            date_idx, :, :
+        ] = cloud_fraction_ice_counts
+        dataset["Cloud_Fraction_Mean"][date_idx, :, :] = cloud_fraction
+        dataset["Cloud_Fraction_Pixel_Counts"][
+            date_idx,
+            :,
+            :,
+        ] = cloud_fraction_counts
+        dataset["Cloud_Fraction_Day_Mean"][date_idx, :, :] = cloud_fraction_day
+        dataset["Cloud_Fraction_Day_Pixel_Counts"][
+            date_idx, :, :
+        ] = cloud_fraction_day_counts
+        dataset["date_idx"][date_idx] = date_idx
+
+    except Exception as e:
+        print(f"Error processing {downloaded_file_name}: {e}")
+
+
+start_date = datetime(year, 1, 1)  # start from Jan 1.
+bash_script_path = "./download_modis_daily.sh"
+destination_path = "./data_tmp/"
+
+# initialize NetCDF file
+nc_filename = f"ccic_modis_data_{year}.nc"
+dataset = nc.Dataset(nc_filename, "w", format="NETCDF4")
+
+dataset.createDimension("date_idx", None)
+date_idxs = dataset.createVariable("date_idx", "i4", ("date_idx",))
+
+dataset.createDimension("date", None)
+dates = dataset.createVariable("date", "i8", ("date",))
+dates.units = "Unix time [s]"
+
+dataset.createDimension("lat", 180)
+lats = dataset.createVariable("lat", "f4", ("lat",))
+lats[:] = np.arange(-89.5, 90.5, 1)
+dataset.createDimension("lon", 360)
+lons = dataset.createVariable("lon", "f4", ("lon",))
+lons[:] = np.arange(-179.5, 180.5, 1)
+
+dataset.createDimension("histo_bins", 16)
+histo_bins = dataset.createVariable(
+    "Cloud_Water_Path_Ice_Histo_Intervals", "i2", ("histo_bins",)
+)
+histo_bins[:] = np.array(
+    [5, 15, 35, 75, 125, 175, 225, 275, 325, 375, 425, 475, 750, 1500, 3000, 5000]
+)
+
+# Create the variable with dimensions (time, lat, lon)
+cloud_water_path_ice_mean = dataset.createVariable(
+    "Cloud_Water_Path_Ice_Mean", "i4", ("date", "lat", "lon")
+)
+cloud_water_path_ice_histo_counts = dataset.createVariable(
+    "Cloud_Water_Path_Ice_Histogram_Counts", "i4", ("date", "histo_bins", "lat", "lon")
+)
+cloud_fraction_ice = dataset.createVariable(
+    "Cloud_Retrieval_Fraction_Ice", "i4", ("date", "lat", "lon")
+)
+cloud_fraction_ice_counts = dataset.createVariable(
+    "Cloud_Retrieval_Fraction_Ice_Pixel_Counts", "i4", ("date", "lat", "lon")
+)
+cloud_fraction = dataset.createVariable(
+    "Cloud_Fraction_Mean", "i4", ("date", "lat", "lon")
+)
+cloud_fraction_counts = dataset.createVariable(
+    "Cloud_Fraction_Pixel_Counts", "i4", ("date", "lat", "lon")
+)
+cloud_fraction_day = dataset.createVariable(
+    "Cloud_Fraction_Day_Mean", "i4", ("date", "lat", "lon")
+)
+cloud_fraction_day_counts = dataset.createVariable(
+    "Cloud_Fraction_Day_Pixel_Counts", "i4", ("date", "lat", "lon")
+)
+
+
+current_date = start_date
+for i in range(n_days):
+    folder_name = current_date.strftime(
+        "%j"
+    )  # day of the year as a folder name (e.g., '001')
+    source_url = f"https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD08_D3/{year}/{folder_name}"
```

Comment:
I went with Aqua because it's on the A-train, so corresponds better with CloudSat. I'm unsure if also including Terra would be better or not. Perhaps a more realistic representation of global trends to have both? I'm not sure how diurnal cycles would affect the comparison with the other datasets though. It again adds to the problem of calculating means from daily means, rather than L2 data, since we would need to average over the Terra mean and the Aqua mean to get an overall mean. I would have to think a bit deeper about whether this would make a big difference.